### PR TITLE
Update Fluxc and Login Libraries to Latest Trunk Version

### PR DESCRIPTION
**Pull request by ParaskP7**
Relates To:
- Login: [Update FluxC to Latest Trunk Hash #92](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/92)
- WCAndroid: [Update login library version #7240](https://github.com/woocommerce/woocommerce-android/pull/7240)

This PR is about updating both, the `FluxC` and `Login` libraries, to their corresponding latest `trunk` version. This is mainly done because of a crash that was identified on the WCAndroid site due to the `RefreshSitesXMLRPCPayload` data class becoming immutable as part of a `FluxC` update. However, the `Login` library was pointing to a previous version of `FluxC`, which didn't have the same change, thus causing that specific flow which was utilizing the previous version of `RefreshSitesXMLRPCPayload` to crash.

This update is done in order to make sure that both, this WPAndroid client and its `Login` library dependency, are pointing to that `FluxC` version where everything works as expected, without crashes or weird behaviors.

-----

👋 @zwarm , I shamelessly added you as an extra reviewer. Can you help @AliSoftware with the smoke testing part, just to verify that everything works as expected before he starts with the `20.6` code freeze on Monday? 🙇 

-----

To test:

- Verify that all the CI checks are successful.
- Smoke test the apps, both WordPress and Jetpack, and verify everything works as expected (like `Login` functionality, `My Site/Reader/Notification` tabs, `Stats/Media/Posts/Pages/Comments/Me/Jetpack` screens, create `Story/Post/Page`, etc).
- Additionally, you can follow the testing instructions within this related [WCAndroid PR](https://github.com/woocommerce/woocommerce-android/pull/7240) PR, and connect with a self-hosted site on a Jetpack connection and verify that the app isn't crashing.

## Regression Notes
1. Potential unintended areas of impact

Since this `FluxC` version comes with lots of small changes, like models becoming immutable (`var` -> `val`, etc), it might be the case that either the app is not complilying (can be easily verified by CI) or the app might be crashing on specific scenarios (thus needing manual smoke testing) due to transitive dependencies on various libraries that point to a different version of `FluxC` (like the `Login` library).

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: build.gradle
Comment (AliSoftware) [inline build.gradle:12]: This matches the sha1 of current Login's `trunk`, so 👍 

But that Login commit… points to `fluxc:trunk-e94126bf9eb942d262768f8851cf2c4980e249fc` [in its own `build.gradle`](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/blob/e11535105725e02dd1e8aa8065314ceb67079278/WordPressLoginFlow/build.gradle#L58), as opposed to pointing to FluxC's latest commit in trunk aka `trunk-eb1d39ad095c01367af03c4259ab93cda14c675e` aka what's also used as `fluxCVersion` in this WPAndroid `build.gradle` on line 24. And FluxC's `e94126b` (which Login points to) is 3 commits behind `eb1d39a`.

While it's not surprising to see Login's pointer to fluxc be behind the FluxC we points to in the clients, especially because FluxC is updated way more often than the Login lib is, I'm a bit surprised by this given that Login and FluxC were just synchronzed a couple of hours ago — in fact, that is the whole points of this update of both Login and FluxC being done in parallel here — and I wouldn't expect FluxC to _already_ have moved ahead much?

So I'm wondering if this is expected to have Login not include those 3 recent commits from FluxC while this was all about synchronizing and updating the version of FluxC that both were pointing to to match 🤔

Comment (AliSoftware) [reply build.gradle:12]: Ok after checking with @ParaskP7 in Slack, this is in fact expected. And the 3 FluxC commits that Login does not include are not relevant to the Login library anyway.

> While it's not surprising to see Login's pointer to fluxc be behind the FluxC we points to in the clients, especially because FluxC is updated way more often than the Login lib is, I'm a bit surprised by this given that Login and FluxC were just synchronzed a couple of hours ago […] and I wouldn't expect FluxC to _already_ have moved ahead much?

So in practice this is just because FluxC already shipped a couple of recent PRs after this synchronization in Login lib happened earlier today. So… FluxC already moving that much ahead already after all 😄

Comment (ParaskP7) [reply build.gradle:12]: 👋 @AliSoftware !

Short answer, yes, this is expected and thank you for asking! 🥇 

- This morning we fixed an issue with the [Login PR](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/92) and then update WCAndroid to point to that with a [WCAndroid PR](https://github.com/woocommerce/woocommerce-android/pull/7240).
- Then, later on today, I merged 3 more PRs on FluxC ([here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2506), [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2504) and [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2507)) and completed the project I was working for this sprint.
- This problem that this PR is trying to fix was on a previous FluxC commit, even before there extra 3 PRs, which got already merged 10 days ago on FluxC (via [this PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2490) and [this commit](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2490/commits/8a96a86fb723285d8a056231ebad6096231aadd9#diff-5c8976cc6b3f8b603138c635d67b4938b5628f9070c4030c5b0b659ba70d40d4)).
- Thus these 3 PRs that were merged afterwards weren’t going to help with this `Login` problem anyway, they are just nice to have on top of what `Login`'s `FluxC` is pointing to.

Just that makes sense to you? 🤔

**Comment by AliSoftware** [comment]
The diff looks good (once my question above wrt to the sha1 was clarified and confirmed that all is expected).

I probably won't have time to smoke-test the app with those changes today tho, as I'm already late to do the final build to submit 20.5 in my day.
So I'll let @zwarm do the smoke-testing, and approve and land the PR if everything looks good to her

File: build.gradle
Comment (AliSoftware) [reply build.gradle:12]: That makes total sense now, thanks for explaining in such details 👍

Comment (ParaskP7) [reply build.gradle:12]: Great, thank you for taking the time to investigate all that! 🥇

**Review by AliSoftware — APPROVED** [review]
Smoke tested both WordPress and Jetpack this morning (as I'll need this PR to land before I can do the code freeze) and didn't find anything unusual, apps worked as expected.

**Pull request merged**
